packages/experimental/Button/src/useButton.ts

```diff
@@ -17,10 +18,16 @@ export const useButton = (props: ButtonProps): ButtonState => {
       accessibilityRole: 'button',
       onAccessibilityTap: props.onAccessibilityTap || props.onClick,
       accessibilityLabel: props.accessibilityLabel || props.content,
+      accessibilityState: getAccessibilityState(!!disabled),
       focusable: true,
       ref: useViewCommandFocus(componentRef),
       onKeyUp: onKeyUp,
     },
     state: pressable.state,
   };
 };
+
+const getAccessibilityState = memoize(getAccessibilityStateWorker);
+function getAccessibilityStateWorker(disabled: boolean) {
+  return { disabled: disabled };
+}
```

Comment:
Why do we need this?

Reply:
To avoid recreating the object on every render if nothing has changed. There's a test for button that checks to make sure that if you rerender the button with no prop changes, there aren't any changes in the props using shallow compare (which would cause a rerender). This was needed to pass that test.

Reply:
I see... could we maybe write a hook that is something like `useButtonAccessbilityProps` or something that handles all the accessibility props? It feels a little weird to have this new pattern for this one prop to achieve the type of caching I thought hooks were supposed to help us avoid. Not a review blocking comment

Reply:
Sure, I'd be happy to try it out. I did try using some hooks (I think?) but those wouldn't pass the test (React.useMemo, React.useCallback).

Reply:
Yeah I'm mostly guessing, obviously you are the one who is actually running the tests and such. I suppose I should look into this worker/memoize pattern

Reply:
I tried wrapping useMemo in another function but it still wouldn't pass the test. I think for now I'll keep this as is. We can refactor and make it look more like a hook once we share this function with other components.